### DISCUSSION TITLE
How to transfer 'Py<PyAny>' to 'Bytes'?

### DISCUSSION THREAD

**Discussion by k82cn** [PyO3/pyo3#3878] [Questions] [answered]
Currently, I got a return value from `dill.dumps()` as `Py<PyAny>`; I'd like to transfer it into `Vec[8]`, so to `bytes::Bytes`. Is there anyway to do that?

Here's the sample code:

```
        let res = task_fn.call(py, args, Some(kwargs))?;

        let dill = py.import("dill")?;
        let dump_fn = dill.getattr("dumps")?;
        let res = PyTuple::new(py, &[res]);

        // How to transfer the return value of `dill.dumps` into `&[8]`?
        let _: Py<PyAny> = dump_fn.call(res, None)?.into();
```

**Comment by davidhewitt** [comment, marked as answer]
At the moment we don't have first-party support for `Bytes` (as per #1992), but that doesn't stop you from doing this conversion yourself via `Bytes` constructors.

Also, in general `Py<T>` is best for storing data and its counterpart `&T` or `Bound<T>` (depending on pre/post 0.21) is easiest to work with for manipulating objects.

I would recommend the following: 

```rust
let any: &PyAny = dump_fn.call(res, None)?; 
let py_bytes: &PyBytes = any.downcast()?;
let bytes: Bytes = Bytes::copy_from_slice(py_bytes.as_bytes());
```

**Reply by k82cn** [reply]
Great. I works in my environment!

I'll share a demo when I integrated it into https://github.com/flame-sh/flame :)

xref: https://github.com/k82cn/rust-learning/tree/main/pyo3
